platforms/software/dependency-management/src/main/java/org/gradle/internal/component/model/AttributeMatchingExplanationBuilder.java

```diff
@@ -23,14 +23,12 @@
 import java.util.Collection;
 
 public interface AttributeMatchingExplanationBuilder {
-    AttributeMatchingExplanationBuilder NO_OP = () -> true;
+    AttributeMatchingExplanationBuilder NO_OP = new AttributeMatchingExplanationBuilder() { };
 
     static AttributeMatchingExplanationBuilder logging() {
         return LoggingAttributeMatchingExplanationBuilder.logging();
     }
 
-    boolean canSkipExplanation();
```

Comment:
Why was this dropped? Do we not want to log in DEBUG every time, for easier debugging?

Reply:
We never _didn't_ pass a `AttributeMatchingExplanationBuilder.logging()` into this method. That method would either return a NO_OP implementation or the `LoggingAttributeMatchingExplanationBuilder.logging()`, both of which always returned true here.

The _reason_ this was changed was to allow `DefaultAttributeMatcher`'s cache to only depend on the key, and not other parameters other than the key. This makes it compatible with a `LoadingCache` where the `computeIfAbsent` callback is specified at the time of constructing the cache instead of each time we access from the cache.

Reply:
Ah, I thought I saw one of them return `false`. Never mind then.